metrics/config_test.go

```diff
@@ -1006,3 +1031,19 @@ func TestStackdriverRecord(t *testing.T) {
 		})
 	}
 }
+
+// setEnvVar sets the value of an env var and returns a function that can be
+// deferred to unset that variable.
+func setEnvVar(t *testing.T, name, val string) (unset func()) {
+	t.Helper()
+
+	if err := os.Setenv(name, val); err != nil {
+		t.Errorf("Failed to set env var %s: %v", name, err)
+	}
+
+	return func() {
+		if err := os.Unsetenv(name); err != nil {
+			t.Logf("Failed to unset env var %q: %s", name, err)
+		}
+	}
+}
```

Comment:
honestly, I'd just go with
```
setenv
defer unsetenv
```
in the test. There is 0 chance this is likely to fail :)

Reply:
The reason why I chose that approach is because the golang-ci linter dislikes unhandled errors.

If you feel strongly about it I can revert, but if we ever run linters in the CI those small quality improvements will matter.

Reply:
eh, sure. :)

Reply:
Reverted in 0ec08e2b, since we don't run linters in Knative.
https://github.com/knative/pkg/blob/0ec08e2b0f1d605c10ca5eba3fed4523e12dca35/metrics/config_test.go#L620-L624